docs/DevGuide/ImplementingVectors.md

```diff
@@ -0,0 +1,254 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# Implementing new Vector types {#implementing_vectors}
+
+\tableofcontents
+
+# Overview of SpECTRE Vectors {#general_structure}
+
+In SpECTRE, sets of contiguous or related data are stored in specializations of
+vector data types. The canonical implementation of this is the `DataVector`,
+which is used for storage of a contiguous sequence of doubles which support a
+wide variety of mathematical operations. However, we support the ability to
+easily generate `DataVector`s for alternative types or which support alternative
+sets of mathematical operations. All such data types are derived classes from
+the class template `VectorImpl`. The remainder of this brief guide gives a
+description of the tools for defining additional Vector types.
+
+For reference, all functions described here can also be found in brief in the
+Doxygen documentation for VectorImpl.hpp, and a simple reference implementation
+can be found in DataVector.hpp and DataVector.cpp.
+
+# The class definition {#class_definition}
+
+SpECTRE vector types inherit from vector types implemented in the
+high-performance arithmetic libarary
+[Blaze](https://bitbucket.org/blaze-lib/blaze). Using this inheritance
+technique, SpECTRE vectors gracefully make use of the math functions defined for
+the Blaze types, but can be customized for the specific needs in SpECTRE
+computations.
+
+The pair of template parameters for `VectorImpl` are the type of the stored data
+(`double` for `DataVector`), and the result type for mathematical operations,
```

Comment:
I'm uncertain about the multiplication, because we'll probably want something representing masks, and I don't think it's yet uncontroversial that masks should be `DataVector`s or `ModalVector`s. I'll change the example to addition for the time being, because I think it is more obvious that it should not be permitted.